### PR TITLE
Use XML output when running tests

### PR DESCRIPTION
**Pull request by OskiKervinen-MF**
Switched to XML output for running tests. It is more unambiguous to parse.

Used the power of the XML output to include the section information in a failure message.

Added basic tests for test execution to make sure the XML approach didn't cause regressions.

Builds on top of #4.

### CODE REVIEW

**Comment by xkbeyer** [comment]
Wow, thanks for the PR so far. I'm going to check it this WE.

File: source.extension.vsixmanifest [hunk 1]
```diff
@@ -1,23 +1,29 @@
...
+        <Icon>Resources\VSPackage1.ico</Icon>
+        <Tags>Unit Test, C++, Visual Studio 2015</Tags>
+    </Metadata>
+    <Installation>
+        <InstallationTarget Id="Microsoft.VisualStudio.Community" Version="[14.0,16.0)" />
+        <InstallationTarget Version="[14.0,16.0)" Id="Microsoft.VisualStudio.Pro" />
+        <InstallationTarget Version="[14.0,16.0)" Id="Microsoft.VisualStudio.Enterprise" />
+    </Installation>
+    <Dependencies>
+        <Dependency Id="Microsoft.Framework.NDP" DisplayName="Microsoft .NET Framework" d:Source="Manual" Version="[4.5,)" />
```
Comment (xkbeyer) [inline source.extension.vsixmanifest:17]: 4.6.1 should be added.

File: source.extension.vsixmanifest [hunk 2]
```diff
@@ -1,23 +1,29 @@
...
+        <Dependency Id="Microsoft.VisualStudio.MPF.14.0" DisplayName="Visual Studio MPF 14.0" d:Source="Installed" Version="[14.0]" />
```
Comment (xkbeyer) [inline source.extension.vsixmanifest:18]: 15.0 should be added.

**Comment by xkbeyer** [comment]
It looks pretty good, however I have some small remarks.
The following test:
`
 TEST_CASE("Fail message test", "[fixture]")
 {
    FAIL("Not implemented!");
 }
`
results in an exception.

The ReferenceCatchProject is still in VS2015
For VS2017 I have to retarget the projects to FrameWorkVersion 4.6.1 and SDK 10.x (I dont have 2015 and SDK 8.1 installed).
If you could fix these I'm fine to merge the PR.

File: source.extension.vsixmanifest (hunk 1 above)
Comment (OskiKervinen-MF) [reply source.extension.vsixmanifest:17]: I do not have a deep understanding of the best practices of these dependency versioning constraints. My understanding was that this expression means "4.5 and above". I made it loose with the idea that it should at least try to run with any future version. Why do you believe a more specific constraint would be better?

Comment (xkbeyer) [reply source.extension.vsixmanifest:17]: You are right. It is better to stay with 4.5. It makes no sense to switch to a newer one if not needed.

**Pull request merged**